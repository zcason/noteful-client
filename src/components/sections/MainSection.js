import React, { Component } from 'react';
import { Link, withRouter } from 'react-router-dom';
import NoteItem from '../NoteItem';
import DefaultContext from '../context/DefaultContext';
import PropTypes from 'prop-types';

class MainSection extends Component {
    static contextType = DefaultContext;

    getNoteList = () => {
        if ( typeof this.props.store !== "object" ) return;
        
        return this.props.store.notes.map(note => {
            return (
                <NoteItem
                    key={note.id}
                    id={note.id}
                    name={note.name}
                    modified={note.modified}
                    folderId={note.folderId}
                    content={note.content}
                    store={this.props.store}
                />
            );
        });
    }
    getBottomContents = () => {
        if ( !this.props.history.location.pathname.includes(`/note/`) ) return (
            <Link 
                to={`/add-note/`}
                className="note--item--add"
            >
                Add Note
            </Link>
        );
        return (<p className="note--item--content">{this.props.store.notes[0].content}</p>);
    }
    render(){
        return (
            <main className="section--main">
                <ul>
                    {this.getNoteList()}
                </ul>
                <div className="section--row">
                    {this.getBottomContents()}
                </div>
            </main>
       );
    }
}

MainSection.propTypes = {
    store: PropTypes.object.isRequired
}

export default withRouter(MainSection);