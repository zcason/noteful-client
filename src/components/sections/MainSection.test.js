import React from 'react';
import { shallow } from 'enzyme'
import toJson from 'enzyme-to-json'
import store from '../../dummy-store';
import MainSection from './MainSection';

describe(`MainSection component`, () => {
    const props = {
      store: store
    }
  
    it('renders MainSection by default', () => {
      const wrapper = shallow(<MainSection />)
      expect(toJson(wrapper)).toMatchSnapshot()
    })
  
    it('renders main section with store when in props', () => {
      const component = shallow(<MainSection {...props} />)
        .find('.section--main')
      expect(toJson(component)).toMatchSnapshot()
    })
  })
  