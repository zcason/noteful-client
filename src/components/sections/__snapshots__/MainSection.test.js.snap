// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`MainSection component renders MainSection by default 1`] = `
<ContextConsumer>
  <Component />
</ContextConsumer>
`;

exports[`MainSection component renders main section with store when in props 1`] = `null`;
